unittests/interface/interface_functions/test_front_end_common_graph_binary_gatherer.py

```diff
@@ -72,8 +71,8 @@ def test_call(self):
         targets = gatherer.__call__(
             placements, graph, _TestExecutableFinder())
         gatherer = LocateExecutableStartType()
-        start_type = gatherer.__call__(graph)
-        self.assertEqual(start_type, ExecutableStartType.RUNNING)
+        start_type = gatherer.__call__(graph, placements)
+        self.assertEqual(start_type.keys()[0], ExecutableType.RUNNING)
```

Comment:
Is this guaranteed to be 0 given it is a set?  Might be better checking size first...

Reply:
its a list, as you stated above about keys(). am testing that it detected the executable type correctly.

Reply:
I think it works in this case - keys() is a call into a dict and the returned list is of unspecified order.  I think that because this test expects only one thing, it is fine but just a warning...

Reply:
I suppose it should then check for size of 1 as well...

Reply:
will add to please

Reply:
rubbered